apps/health_campaign_field_worker_app/lib/blocs/search_households/search_households.dart

```diff
@@ -225,75 +223,61 @@ class SearchHouseholdsBloc
+    final List<HouseholdModel> proximityBasedHouseholdsResults =
+        await addressRepository.searchHouseHoldbyAddress(AddressSearchModel(
       latitude: event.latitude,
       longitude: event.longititude,
       maxRadius: event.maxRadius,
+      offset: event.offset,
+      limit: event.limit,
     ));
 
     // Extract individual IDs from proximity-based individual results.
-    final indIds = proximityBasedIndividualResults
+    final househHoldIds = proximityBasedHouseholdsResults
         .map((e) => e.clientReferenceId)
         .toList();
 
-    // Search for individual results using the extracted IDs and search text.
-    final List<IndividualModel> indResults = await individual.search(
-      IndividualSearchModel(
-        clientReferenceId: indIds,
-      ),
+    final List<HouseholdMemberModel> householdMembers =
+        await fetchHouseholdMembersBulk(
+      null,
+      househHoldIds,
     );
 
-    // Search for individual results based on the search text only.
+    final List<String> individualClientReferenceIds = householdMembers
+        .map((e) => e.individualClientReferenceId.toString())
+        .toList();
 
-    // Initialize a list to store household members.
-    final householdMembers = <HouseholdMemberModel>[];
-
-    // Determine the source of results based on proximity and beneficiary type.
-    final r = beneficiaryType != BeneficiaryType.individual
-        ? proximityBasedResults
-        : indResults;
-
-    // Iterate through the results and retrieve household members.
-    for (final element in r) {
-      // Determine the type of household members based on proximity and beneficiary type.
-      final members = beneficiaryType != BeneficiaryType.individual
-          ? await fetchHouseholdMembers(
-              (element as HouseholdModel).clientReferenceId,
-              null,
-              true,
-            )
-          : await fetchHouseholdMembers(
-              null,
-              (element as IndividualModel).clientReferenceId,
-              true,
-            );
-
-      // Retrieve all household members associated with each member.
-      for (final member in members) {
-        final allHouseholdMembers = await householdMember.search(
-          HouseholdMemberSearchModel(
-            householdClientReferenceId: member.householdClientReferenceId,
-          ),
-        );
+    final List<IndividualModel> individuals = await individual.search(
+      IndividualSearchModel(clientReferenceId: individualClientReferenceIds),
+    );
 
-        householdMembers.addAll(allHouseholdMembers);
-      }
+    final projectBeneficiaries = await fetchProjectBeneficiary(
+      beneficiaryType != BeneficiaryType.individual
+          ? househHoldIds
+          : individualClientReferenceIds,
+    );
+
+    List<SideEffectModel> sideEffects = [];
+    List<ReferralModel> referrals = [];
+    List<TaskModel> tasks = [];
+    if (projectBeneficiaries.isEmpty) {
```

Comment:
Shouldn't this be projectBeneficiaries.isNotEmpty ?? @naveen-egov it is there below as well